src/spatialdata/_core/spatialdata.py

```diff
@@ -315,6 +320,52 @@ def get_instance_key_column(table: AnnData) -> pd.Series:
             return table.obs[instance_key]
         raise KeyError(f"{instance_key} is set as instance key column. However the column is not found in table.obs.")
 
+    @staticmethod
+    def set_image_channel_names(element: DataArray | DataTree, channel_names: str | list[str]) -> DataArray | DataTree:
+        """Set the channel names for a image `SpatialElement` in the `SpatialData` object.
+
+        Parameters
+        ----------
+        element
+            The image `SpatialElement` or parsed `ImageModel`.
+        channel_names
+            The channel names to be assigned to the c dimension of the image `SpatialElement`.
+
+        Returns
+        -------
+        element
+            The image `SpatialElement` or parsed `ImageModel` with the channel names set to the `c` dimension.
+        """
+        channel_names = channel_names if isinstance(channel_names, list) else [channel_names]
+        model = get_model(element)
+        if model in [Image2DModel, Image3DModel]:
+            channel_names = _check_match_length_channels_c_dim(element, channel_names, model().dims.dims)  # type: ignore[union-attr]
+            if isinstance(element, DataArray):
+                element = element.assign_coords(c=channel_names)
+            else:
+                element = element.map_over_datasets(_assign_multiscale_coords, {"c": channel_names})
+        else:
+            raise TypeError(f"Model `{model}` does not support setting channel names.")
+
+        return element
+
+    def set_sdata_image_channel_names(self, element_name: str, channel_names: str | list[str]) -> None:
```

Comment:
I would remove this method because we started following the philosophy of separating the modification of data in-memory and the on-disk write, as with transformations. 

So now the workflow would be to call `set_channels()` (or `set_channel_names()` depending on how we call it) and then manually call `sdata.write_channel_names()` or `sdata.write_metadata()`.

Reply:
I would still have this as a method, but then indeed separating the in memory modification and the write on disk and adding an argument to this function write which is a bool